Apps/BMSQueryApp/app/src/main/java/uk/ac/cam/cares/jps/bmsqueryapp/EquipmentInstanceActivity.java

```diff
@@ -114,13 +114,25 @@ public void onTabSelected(TabLayout.Tab tab) {
         });
     }
 
-    private List<EditableAttribute> getEditableAttributeList(String type) {
+    private List<EditableAttribute> getEditableAttributeList(String type, String equipmentIRI) {
         // determine the list of editable attribute based on the equipment type
         if (type.equals("https://www.theworldavatar.com/kg/ontobms/WalkInFumeHood")) {
             return new ArrayList<>();
         } else if (type.equals("https://www.theworldavatar.com/kg/ontodevice/SmartSensor") || type.equals("https://w3id.org/s3n/SmartSensor")) {
             ArrayList<EditableAttribute> attributes = new ArrayList<>();
-            attributes.add(new EditableAttribute("https://www.theworldavatar.com/kg/ontodevice/V_Setpoint-01-Temperature", "Temperature", "double", "°C"));
+            attributes.add(new EditableAttribute("https://www.theworldavatar.com/kg/ontodevice/V_Setpoint-01-Temperature", "Temperature Setpoint", "double", "°C"));
+            return attributes;
+        } else if (type.equals("https://www.theworldavatar.com/kg/ontobms/ExhaustVAV") && equipmentIRI.equals("https://www.theworldavatar.com/kg/ontobms/VAV_E7_01_d00bd62d-9f86-4f01-8940-c52a62269f4c")) {
+            ArrayList<EditableAttribute> attributes = new ArrayList<>();
+            attributes.add(new EditableAttribute("https://www.theworldavatar.com/kg/ontobms/V_VAV_E-7-1_FlowSP_CARES", "Airflow Setpoint", "double", "m3/h"));
+            return attributes;
+        } else if (type.equals("https://www.theworldavatar.com/kg/ontobms/ExhaustVAV") && equipmentIRI.equals("https://www.theworldavatar.com/kg/ontobms/VAV_E7_02_e4ac5287-26d8-4638-9ba2-74b9e9994814")) {
+            ArrayList<EditableAttribute> attributes = new ArrayList<>();
+            attributes.add(new EditableAttribute("https://www.theworldavatar.com/kg/ontobms/V_VAV_E-7-2_FlowSP_CARES", "Airflow Setpoint", "double", "m3/h"));
+            return attributes;
+        } else if (type.equals("https://www.theworldavatar.com/kg/ontobms/CanopyHood") && equipmentIRI.equals("https://www.theworldavatar.com/kg/ontobms/CH-7-7_CAV_90f2f336-03f7-4c0e-80a0-6d85873e42a8")) {
+            ArrayList<EditableAttribute> attributes = new ArrayList<>();
+            attributes.add(new EditableAttribute("https://www.theworldavatar.com/kg/ontobms/V_CAV_E-7-7_FlowSP_CARES", "Airflow Setpoint", "double", "m3/h"));
```

Comment:
May be better if can simplify this piece of code. The current code looks a bit messy. A possible way is to 
``` java
else if (VAV type || CanopyHood) {
   ...
   attributes.add(new EditableAttribute(equipmentIRIToDataIRI.get(equipmentIRI), "Airflow Setpoint", "double", "m3/h"));
   // should handle if equipmentIRI not in the dict case 
   ...
}

// equipmentIRIToDataIRI is a dictionary at class level that store equipmentIRI and the corresponding DataIRI
```

Reply:
Code has been simplified with the addition of IRIMapping class that constructs a map containing the IRIs